libflux/core/src/semantic/flatbuffers/mod.rs

```diff
@@ -714,28 +714,10 @@ impl<'a> semantic::walk::Visitor<'_> for SerializingVisitor<'a> {
             }
 
             walk::Node::TestCaseStmt(test) => {
-                let assignment = {
-                    match v.stmts.pop() {
-                        Some((union, fbsemantic::Statement::NativeVariableAssignment)) => {
-                            Some(WIPOffset::new(union.value()))
-                        }
-                        _ => {
-                            v.err = Some(String::from(
-                                "failed to pop assignment statement from stmt vector",
-                            ));
-                            return;
-                        }
-                    }
-                };
-
-                let test = fbsemantic::TestCaseStatement::create(
-                    &mut v.builder,
-                    &fbsemantic::TestCaseStatementArgs { loc, assignment },
-                );
-                v.stmts.push((
-                    test.as_union_value(),
-                    fbsemantic::Statement::TestCaseStatement,
-                ));
+                // TestCase statements should be transformed before the semantic phase. Even without
+                // the explicit panic here, an error would occur because the block has no function
+                // scope.
+                panic!("TestCaseStmt is not supported in semantic analysis.");
```

Comment:
My instinct said "We should explode if we get to semantic analysis and we still have a `testcase` statement in there." Still, I sat down and tried to implement it. Unfortunately, we're doing something with `Block` in the `testcase` statement that doesn't really fit the semantic analysis, in that it's not connected to a `FunctionExpression`. I didn't feel confident that allowing that wouldn't have unintended consequences (in fact, it would likely have many of them).

Reply:
Hmm, in that case, can we add an informative error to the code that does the analysis instead of panicking here? I can imagine other uses of `walk` other than the type checking pass that we do.

Also, maybe I'm missing something, but do we need a semantic graph representation of `testcase` at all? It looks like it's transformed away before we do anything with it.

Reply:
I've removed all semantic support for `testcase`